src/Pages/Signup/Signup.js

```diff
@@ -0,0 +1,132 @@
+import "./Signup.css";
+import axios from "axios";
+import { useNavigate } from "react-router-dom";
+import { useAuthContext } from "../../Context/AuthContext";
+import { useToast } from "../../Context/ToastContext";
+import { validateInput } from "../../Utils/validateInput";
+
+export const Signup = () => {
+  let navigate = useNavigate();
+  const { auth_state, setAuthState } = useAuthContext();
+  const { user } = auth_state;
+  const { firstName, lastName, email, password, confirmPassword } = user;
+  const { handleaddtoast } = useToast();
+  const signupHandle = async () => {
+    try {
+      let validate_result = validateInput(
+        email,
+        password,
+        confirmPassword,
+        firstName,
+        lastName
+      );
+
+      if (validate_result === "OK") {
+        const response = await axios.post(`/api/auth/signup`, {
+          firstName: firstName,
+          lastName: lastName,
+          email: email,
+          password: password,
+        });
+        localStorage.setItem("token", response.data.encodedToken);
+        setAuthState({ type: "TOKEN", payload: response.data.encodedToken });
+        handleaddtoast({
+          message: `Welcome ${firstName}`,
+          type: "alert-success",
+        });
+        navigate("/");
```

Comment:
use the "from" state to naviagate the user to the clicked page.

Reply:
Okay..Will Use it..